### PR TITLE
run machine scale up/down case in all qe jobs

### PR DESCRIPTION
**Pull request by sunzhaohua2**
Machineset scale up/down is a very important case, as it is destructive, it is often not run in some clusters, To make sure it can be run in every cluster, this will add it to non-destructive jobs, run it after all cases is finished.
@jhou1 @miyadav @huali9 PTAL

### CODE REVIEW

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-imdsv2-fips-f14

File: ci-operator/step-registry/openshift-extended/test/supplementary/openshift-extended-test-supplementary-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    documentation: |
+      It is additional scenario to combine the TEST_SCENARIOS to make final scenarios.
+  - name: TEST_FILTERS_SUPPLEMENTARY
+    default: "~ChkUpgrade&;~DisconnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&"
+    documentation: |
+      It is filters to select case based on some case attribution which is in case title.
+  - name: FILTERS_ADDITIONAL_SUPPLEMENTARY
+    default: ""
+    documentation: |
+      It is additional filters to combine the TEST_FILTERS to make final filters.
```
Comment (kuiwang02) [inline ci-operator/step-registry/openshift-extended/test/supplementary/openshift-extended-test-supplementary-ref.yaml:25]: @sunzhaohua2 
change to `It is additional filters to combine the FILTERS_ADDITIONAL_SUPPLEMENTARY to make final filters.`

File: ci-operator/step-registry/openshift-extended/test/supplementary/openshift-extended-test-supplementary-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  env:
+  - name: TEST_SCENARIOS_SUPPLEMENTARY
+    default: |
+        "25436"
+    documentation: |-
+      Select the test case. It's a RE pattern and is passed to the 'grep -E'
+  - name: TEST_ADDITIONAL_SUPPLEMENTARY
+    default: ""
+    documentation: |
+      It is additional scenario to combine the TEST_SCENARIOS to make final scenarios.
```
Comment (kuiwang02) [inline ci-operator/step-registry/openshift-extended/test/supplementary/openshift-extended-test-supplementary-ref.yaml:17]: @sunzhaohua2 
change to `It is additional scenario to combine the TEST_SCENARIOS_SUPPLEMENTARY to make final scenarios.`

**Comment by kuiwang02** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-ipi-byo-route53-f14

**Comment by kuiwang02** [comment]
/pj-rehearse  periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-shared-vpc-phz-f14

**Comment by kuiwang02** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-ipi-byo-route53-f14

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-imdsv2-fips-f14

File: ci-operator/step-registry/openshift-extended/test/supplementary/openshift-extended-test-supplementary-commands.sh [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+}
+function check_case_selected {
+    found_ok=$1
+    if [ "W${found_ok}W" == "W0W" ]; then
+        echo "find case"
+    else
+        echo "do not find case"
+    fi
+}
+run
```
Comment (kuiwang02) [inline ci-operator/step-registry/openshift-extended/test/supplementary/openshift-extended-test-supplementary-commands.sh:450]: @sunzhaohua2 need to remove this warning with adding newline without space

File: ci-operator/step-registry/openshift-extended/test/supplementary/openshift-extended-test-supplementary-ref.metadata.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+			"kuiwang02"
+		],
+		"reviewers": [
+			"jianzhangbjz",
+			"kuiwang02",
+			"xia-zhao-rh"
+		]
+	}
+}
+
```
Comment (kuiwang02) [inline ci-operator/step-registry/openshift-extended/test/supplementary/openshift-extended-test-supplementary-ref.metadata.json:15]: @sunzhaohua2 need to remove the last line to resolve step-registry-metadata failire. and then we get warning here we have to accept.

**Comment by kuiwang02** [comment]
I am ok from golang's view

File: ci-operator/step-registry/openshift-extended/test/supplementary/openshift-extended-test-supplementary-commands.sh [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+    mapfile -t test_suite_failures < <(grep -r -E 'testsuite.*failures="[1-9][0-9]*"' "${ARTIFACT_DIR}" | grep -o -E 'failures="[0-9]+"' | sed -E 's/failures="([0-9]+)"/\1/')
+    failures=0
+    for (( i=0; i<${#test_suite_failures[@]}; ++i ))
+    do
+        let failures+=${test_suite_failures[$i]}
+    done
+    if [ $((failures)) == 0 ]; then
+        echo "All tests have passed"
+    else
+        echo "${failures} failures in openshift-extended-test" | tee -a "${SHARED_DIR}/openshift-e2e-test-qe-report-openshift-extended-test-failures"
```
Comment (liangxia) [inline ci-operator/step-registry/openshift-extended/test/supplementary/openshift-extended-test-supplementary-commands.sh:323]: The file `openshift-e2e-test-qe-report-openshift-extended-test-failures` is used to track the number of test failures  for ginkgo tests. Using the same file name will overwrite the number of failures in step `openshift-extended-test`

Comment (kuiwang02) [reply ci-operator/step-registry/openshift-extended/test/supplementary/openshift-extended-test-supplementary-commands.sh:323]: @sunzhaohua2 @liangxia 
so, change it to `echo "${failures} failures in openshift-extended-test-supplementary" | tee -a "${SHARED_DIR}/openshift-e2e-test-qe-report-openshift-extended-test-failures"`?

**Pull request merged**